doc/whatsnew/fragments/4741.feature

```diff
@@ -0,0 +1,10 @@
+A new `json2` reporter has been added. It features a more enriched output that is
+easier to parse and provides more info.
+
+Compared to `json` the only changes are that messages are now under the `"messages"`
+key and that `"message-id"` now follows the camelCase convention and is renamed to
+`"messageId"`.
+
```

Comment:
There's something about the score to announce too, this is what prompted the change in the first place.